parsl/monitoring/db_manager.py

```diff
@@ -302,16 +323,20 @@ def start(self, priority_queue, node_queue, resource_queue) -> None:
 
         while (not self._kill_event.is_set() or
                self.pending_priority_queue.qsize() != 0 or self.pending_resource_queue.qsize() != 0 or
-               priority_queue.qsize() != 0 or resource_queue.qsize() != 0):
+               self.pending_node_queue.qsize() != 0 or self.pending_block_queue.qsize() != 0 or
+               priority_queue.qsize() != 0 or resource_queue.qsize() != 0 or
+               node_queue.qsize() != 0 or block_queue.qsize() != 0):
 
             """
             WORKFLOW_INFO and TASK_INFO messages (i.e. priority messages)
 
             """
-            logger.debug("""Checking STOP conditions: {}, {}, {}, {}, {}""".format(
+            logger.debug("""Checking STOP conditions: {}, {}, {}, {}, {}, {}, {}, {}, {}""".format(
                               self._kill_event.is_set(),
                               self.pending_priority_queue.qsize() != 0, self.pending_resource_queue.qsize() != 0,
-                              priority_queue.qsize() != 0, resource_queue.qsize() != 0))
+                              self.pending_node_queue.qsize() != 0, self.pending_block_queue.qsize() != 0,
```

Comment:
Those checks (`node_queue`) were missed in previous PRs. I am adding all of the checks here

Reply:
so the problem this is fixing is that some node messages might not be written to the database at the end of db manager, if there aren't other messages around keeping the db manager alive?

Reply:
Yes, you are correct. Previously it was not a problem because node messages were only written when a node registered with interchange. But now we also log when a node becomes inactive in PR #1876